lib/facility.rb

```diff
@@ -1,14 +1,72 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name,
+              :address,
+              :phone,
+              :hours,
+              :holidays_closed,
+              :services,
+              :registered_vehicles,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(branch_info)
+    @name = branch_info[:name]
+    @address = branch_info[:address]
+    @phone = branch_info[:phone]
+    @hours = branch_info[:hours]
+    @holidays_closed = branch_info[:holidays_closed]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
-end
+
+  def register_vehicle(vehicle)
+    unless vehicle.registered? || !services.include?('Vehicle Registration')
+      registered_vehicles << vehicle
+      vehicle.update_registration_date
+      vehicle.update_plate_type
+      add_collected_fees(vehicle)
+    end
+  end
+
+  def add_collected_fees(vehicle)
+    if vehicle.antique?
+      @collected_fees += 25
+    elsif vehicle.electric_vehicle?
+      @collected_fees += 200
+    else
+      @collected_fees += 100
+    end
```

Comment:
One way to get ride of if/else statements is to make a "lookup table" using a hash. If you have the plate type as the key and the associated value as the fee, encapsulate that in a method, then you can use that to get rid of if statements like this.